### DISCUSSION TITLE
manufacturer specific cluster dev help

### DISCUSSION THREAD

**Discussion by DanielNagy** [Koenkk/zigbee2mqtt#11351] [Devices]
Hello, I'm looking for a little assistance.
I previously added support for a device using a external converter .js based on these instructions https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#instructions

CH2AX_SWITCH_1.js

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['CH2AX/SWITCH/1'],
    model: '41E2PBSWMZ/356PB2MBTZ',
    vendor: 'Schneider Electric',
    description: 'Wiser 40/300-Series Module Switch 2A',
    extend: extend.switch(),
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff']);
        await reporting.onOff(endpoint);
    }
};

module.exports = definition;
```
I've managed to create a pull request based on the above information which is now merged -  https://github.com/Koenkk/zigbee-herdsman-converters/pull/3737

The device has detected an endpoint 21, with a input cluster of "elkoSwitchConfigurationClusterServer", which is Manufacturer specific cluster of 0xff17. Obviously this is a Schneider not Elko also.

I've got a wireshark details of exactly what Cluster/Manu code / Command / Attributes I want to add 

```
IEEE 802.15.4 Data, Dst: 0x9c35, Src: 0x0000
ZigBee Network Layer Data, Dst: 0x9c35, Src: 0x0000
ZigBee Application Support Layer Data, Dst Endpt: 21, Src Endpt: 1
    Frame Control Field: Data (0x40)
    Destination Endpoint: 21
    Cluster: Manufacturer Specific (0xff17)
    Profile: Home Automation (0x0104)
    Source Endpoint: 1
    Counter: 142
ZigBee Cluster Library Frame, Mfr: Schneider Electric (0x105e), Command: Write Attributes, Seq: 59
    Frame Control Field: Profile-wide (0x04)
    Manufacturer Code: Schneider Electric (0x105e)
    Sequence Number: 59
    Command: Write Attributes (0x02)
    Attribute Field, Enum8: 1
        Attribute: 0x0000
        Data Type: 8-Bit Enumeration (0x30)
        Uint8: 1 (0x01)
```

The question is, how do I continue to locally develop this within the constraints of my HA/zigbee2mqtt and this a single file. 
It appears I need to modify zcl/definition/cluster.ts with the specifics of this cluster 0xff17 which isn't accessible due to being Homeassistant/docker, and add the appropriate commands. 
Am I somehow able to add these custom cluster settings to the single external convertor file I've used above?

**Comment by Suxsem** [comment]
Any help?

**Comment by roastcom** [comment]
Just pinching in, I'm also looking for a solution to communicate with a custom Zigbee cluster.

One solution to handle custom cluster IDs involves patching clusters.js (https://github.com/Koenkk/zigbee2mqtt/issues/18321#issuecomment-1886709220) but that only goes to IDs, from what I understand, nothing on making a non-to-spec cluster structures, I'm afraid.
